index.js

```diff
@@ -52,13 +52,16 @@ module.exports = class SubscriptionDedupe {
 
   unsubscribe(topic/*: string */) {
     let existing = this.subscriptions[topic];
+    let promise
 
     if (existing) {
       existing.refCount--;
       if (existing.refCount === 0) {
-        this.options.onUnsubscribe(topic);
+        promise = existing.promise.then(() => this.options.onUnsubscribe(topic));
```

Comment:
This can still have a race-condition if we rapidly reopen the subscription, will update.

Reply:
Thanks. Could you add a test as well?